core/sql/cli/CliExtern.cpp

```diff
@@ -6316,7 +6316,8 @@ Lng32 SQL_EXEC_DeleteHbaseJNI()
       threadContext->incrNumOfCliCalls();
 
       HBaseClient_JNI::deleteInstance();
-      HiveClient_JNI::deleteInstance();
+      // The Hive client persists across connections
+      // HiveClient_JNI::deleteInstance();
```

Comment:
Sorry, I added a couple of comments to the JIRA, TRAFODION-1910 (https://issues.apache.org/jira/browse/TRAFODION-1910), should have added them here.

Let me try to summarize: The comments I made try to explain that we _could_ treat HiveClient_JNI and HBaseClient_JNI differently. Suresh is asking why we _should_ do that.

So, I think what you are telling me is that I should write a new fix that treats the two objects in a similar way. Hope to have that ready soon. Thanks for your input.